src/Microsoft.OData.Core/WriterValidationUtils.cs

```diff
@@ -32,11 +38,27 @@ internal static void ValidateMessageWriterSettings(ODataMessageWriterSettings me
 
             if (messageWriterSettings.BaseUri != null && !messageWriterSettings.BaseUri.IsAbsoluteUri)
             {
+#if NETCOREAPP3_1_OR_GREATER || NETSTANDARD2_0_OR_GREATER

```

Comment:
I think we should encapsulate this in a method to avoid the `#if NETCORE...` everywhere.